### DISCUSSION TITLE
使用 Typography preset 时，相关的 CSS 变量没有被输出

### DISCUSSION THREAD

**Discussion by bohecola** [unocss/unocss#3386] [Q&A] [answered]
使用 Typography preset 这个预设，--un-prose-* 这些变量没有被输出，导致一些样式获取不到变量，从而不生效。请问该如何配置，让这些变量可以输出出来

![Screenshot 2023-11-28 at 04 16 34](https://github.com/unocss/unocss/assets/68556445/dc6474b6-0851-4f84-b02f-44ad9708cc99)

例如写了一个表格，但是里面元素的边框颜色变量 --un-prose-borders 没有被输出，导致 var(--un-prose-borders) 获取不到

https://unocss.dev/play/?html=DwEwlgbgBAxgNgQwM5ILwCIAOAnA9kgU3QD4AoKKYAFwQCM4CyKLqALAhEJ5lq7bnr1bEAcggC2BYAHoqw8oKHEAggHMps%2BYpl8BlTRy4LetXCACee3v2ODqXACq5xyHUcW8uARgAcbq-q6tjqmFtw6dAxkMuAQZEA&config=PTAEFMGcBsEsDsAuBaAJrSBDARtcz5wAPFOQ0eAe2QCcpEbYBjRcVZWAWwAdKbFIAKC69%2BoAN6DQoVOABmCcAGFK8BQHMANFNDc6kcIgAqAT17qambgAsT26XqiGAgogaxsAV0Sw5dnY4GiACSTKqQ9rr6hgCqVNoAvqByNJScoADknlRMkJAZgoLEoogy8pie0KWyCoQqarDqABSS0jSVUABcoADaOtI9GUyekIhptB0ZmhKgYdB83Rl0qBmgCQC6kZs6kNZ8iMMC3a3SmcOj47v7hxmLrCTI0Oqg9yh8mPDq4KB7AG7gNE6r2QrEw0CmOgSkUChkg3T6pyiTkQcUoTQAlJEHNFjGZKBYrLYMVikUFXO4vD4-MT%2BqTDKFwi1adJIEwweBugBGAB0ACYSdImKh4ItrG5uHCQFBONzdsAIYiEpidNslYIgA&css=PQKgBA6gTglgLgUzAYwK4Gc4HsC2YDCAyoWABYJQIA0YAhgHYAmYcUD6AZllDhWOqgAOg7nAB0YAGLcwCAB60cggDYIAXGBDAAUKDBi0mXGADe2sGC704AWgDuCGAHNScDQFYADJ4Dc5sAACtMLKAJ5gggCMLPK2ABR2pPBIcsoAlH4WAEa0yADWTlBYqEw2yFjK3Bpw5LxxAOTllVDoYpSMYgAs3vUZ2gC%2BmsBAA&options=N4IgLgTghgdgzgMwPYQLYAkyoDYgFyQCuApgL5A

**Comment by zyyv** [comment, marked as answer]
你需要指定你的 typography 的主题色，例如 `prose-red` 来为你生成颜色。 https://unocss.dev/presets/typography#colors
[Playground](https://unocss.dev/play/?html=DwEwlgbgBAxgNgQwM5ILwCIAOAnA9kgUyh3wIFoBzbBAT3QD4AoKKYAFwQCM4CmWX2ACwIIQffgLbZxEyYPoA5BAFsCwAPRt5zWXPoBBCms3bdGqTNYmRYnZM64QNS5Ol3Z7MQBVcy5OdtdSTEARgAOAJcrC3dzBydxcy4eJg1wCCYgA&config=PTAEFMGcBsEsDsAuBaAJrSBDARtcz5wAPFOQ0eAe2QCcpEbYBjRcVZWAWwAdKbFIAKC69%2BoAN6DQoVOABmCcAGFK8BQHMANFNDc6kcIgAqAT17qambgAsT26XqiGAgogaxsAV0Sw5dnY4GiACSTKqQ9rr6hgCqVNoAvqByNJScoADknlRMkJAZgoLEoogy8pie0KWyCoQqarDqABSS0jSVUABcoADaOtI9GUyekIhptB0ZmhKgYdB83Rl0qBmgCQC6kZs6kNZ8iMMC3a3SmcOj47v7hxmLrCTI0Oqg9yh8mPDq4KB7AG7gNE6r2QrEw0CmOgSkUChkg3T6pyiTkQcUoTQAlJEHNFjGZKBYrLYMVikUFXO4vD4-MT%2BqTDKFwi1adJIEwweBugBGAB0ACYSdImKh4ItrG5uHCQFBONzdsAIYiEpidNslYIgA&css=PQKgBA6gTglgLgUzAYwK4Gc4HsC2YDCAyoWABYJQIA0YAhgHYAmYcUD6AZllDhWOqgAOg7nAB0YAGLcwCAB60cggDYIAXGBDAAUKDBi0mXGADe2sGC704AWgDuCGAHNScDQFYADJ4Dc5sAACtMLKAJ5gggCMLPK2ABR2pPBIcsoAlH4WAEa0yADWTlBYqEw2yFjK3Bpw5LxxAOTllVDoYpSMYgAs3vUZ2gC%2BmsBAA&options=N4IgLgTghgdgzgMwPYQLYAkyoDYgFwJTZwCmAvkA)

或者使用 [cssExtend](https://unocss.dev/presets/typography#cssextend) 覆盖原有的预设样式

**Reply by bohecola** [reply]
感谢，明白了
